installer/src/main/bin/shared.cmd

```diff
@@ -0,0 +1,250 @@
+@ECHO OFF
+@rem **************************************************************************
+@rem shared.cmd
+@rem
+@rem Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+@rem Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+@rem
+@rem     NAME
+@rem       shared.cmd - shared script for use with WebLogic Deploy Tooling.
+@rem
+@rem     DESCRIPTION
+@rem       This script contains shared functions for use with WDT scripts.
+@rem
+
+CALL %*
+
+GOTO :ENDFUNCTIONS
+
+:javaSetup
+    @rem Make sure that the JAVA_HOME environment variable is set to point to a
+    @rem JDK with the specified level or higher (and that it isn't OpenJDK).
+
+    SET MIN_JDK_VERSION=%1
+
+    IF NOT DEFINED JAVA_HOME (
+      ECHO Please set the JAVA_HOME environment variable to point to a Java 8 installation >&2
+      EXIT /B 2
+    ) ELSE (
+      IF NOT EXIST "%JAVA_HOME%" (
+        ECHO Your JAVA_HOME environment variable to points to a non-existent directory: %JAVA_HOME% >&2
+        EXIT /B 2
+      )
+    )
+    FOR %%i IN ("%JAVA_HOME%") DO SET JAVA_HOME=%%~fsi
+    IF %JAVA_HOME:~-1%==\ SET JAVA_HOME=%JAVA_HOME:~0,-1%
+
+    IF EXIST %JAVA_HOME%\bin\java.exe (
+      FOR %%i IN ("%JAVA_HOME%\bin\java.exe") DO SET JAVA_EXE=%%~fsi
+    ) ELSE (
+      ECHO Java executable does not exist at %JAVA_HOME%\bin\java.exe does not exist >&2
+      EXIT /B 2
+    )
+
+    FOR /F %%i IN ('%JAVA_EXE% -version 2^>^&1') DO (
+      IF "%%i" == "OpenJDK" (
+        ECHO JAVA_HOME %JAVA_HOME% contains OpenJDK^, which is not supported >&2
+        EXIT /B 2
+      )
+    )
+
+    FOR /F tokens^=2-5^ delims^=.-_^" %%j IN ('%JAVA_EXE% -fullversion 2^>^&1') DO (
+      SET "JVM_FULL_VERSION=%%j.%%k.%%l_%%m"
+      SET "JVM_VERSION_PART_ONE=%%j"
+      SET "JVM_VERSION_PART_TWO=%%k"
+    )
+
+    SET JVM_SUPPORTED=1
+    IF %JVM_VERSION_PART_ONE% LEQ 1 (
+        IF %JVM_VERSION_PART_TWO% LSS %MIN_JDK_VERSION% (
+            SET JVM_SUPPORTED=0
+            IF %JVM_VERSION_PART_TWO% LSS 7 (
+                ECHO You are using an unsupported JDK version %JVM_FULL_VERSION% >&2
+            ) ELSE (
+                ECHO JDK version 1.8 or higher is required when using encryption >&2
```

Comment:
Isn't the encrypt jdk check already in the encrypt model? Is this print necessary?

Reply:
encryptModel just passes in 8 as the required version, and this function logs and returns the exit code if the version is lower.